src/Subscriber/ProxySubscriber.php

```diff
@@ -108,66 +100,29 @@ public function proxy(RequestEvent $event): void
+        $headers = array_merge_recursive($source->getHeaders(), $event->getRequest()->headers->all());
+
+        $endpoint = '';
+        if (isset($headers['x-endpoint'][0]) === true) {
+            $endpoint = trim($headers['x-endpoint'][0], '/');
         }
 
-        $endpoint = rtrim($endpoint, '/');
+        $data['path']['{route}'] = $endpoint;
 
-        $method = ($headers['x-method'][0] ?? $event->getRequest()->getMethod());
-        unset($headers['authorization']);
+        $data['method'] = ($headers['x-method'][0] ?? $event->getRequest()->getMethod());
         unset($headers['x-endpoint']);
         unset($headers['x-method']);
 
-        $url = \Safe\parse_url($source->getLocation());
-
-        try {
-            if ($url['scheme'] === 'http' || $url['scheme'] === 'https') {
-                $result = $this->callService->call(
-                    $source,
-                    $endpoint,
-                    $method,
-                    [
-                        'headers' => $headers,
-                        'query'   => $this->requestService->realRequestQueryAll($event->getRequest()->getQueryString()),
-                        'body'    => $event->getRequest()->getContent(),
-                    ]
-                );
-            } else {
-                $result = $this->fileSystemService->call($source, $endpoint);
-                $result = new \GuzzleHttp\Psr7\Response(200, [], $this->serializer->serialize($result, 'json'));
-            }
-        } catch (ServerException | ClientException | RequestException $exception) {
-            $statusCode = ($exception->getCode() ?? 500);
-            if (method_exists(get_class($exception), 'getResponse') === true && $exception->getResponse() !== null) {
-                $body       = $exception->getResponse()->getBody()->getContents();
-                $statusCode = $exception->getResponse()->getStatusCode();
-                $headers    = $exception->getResponse()->getHeaders();
-            }
-
-            $content = $this->serializer->serialize(
-                [
-                    'Message' => $exception->getMessage(),
-                    'Body'    => ($body ?? "Can\'t get a response & body for this type of Exception: ").get_class($exception),
-                ],
-                'json'
-            );
-
-            $result = new \GuzzleHttp\Psr7\Response($statusCode, $headers, $content);
-
-            // If error catched dont pass event->getHeaders (causes infinite loop)
-            $wentWrong = true;
-        }//end try
-
-        $event->setResponse(new Response($result->getBody()->getContents(), $result->getStatusCode(), !isset($wentWrong) ? $result->getHeaders() : []));
+        $data['headers']     = $headers;
+        $data['querystring'] = $event->getRequest()->getQueryString();
+        $data['crude_body']  = $event->getRequest()->getContent();
+
+        $event->setResponse($this->requestService->proxyHandler($data, [], $source));
```

Comment:
Was this also tested with the proxy-on-source functionality?

Reply:
I have tested the endpoint for this with a source "/admin/sources/id/proxy". And got a response back from the Source as expected.
If that is wat you mean?

Reply:
Yup, thanks for confirming that was tested :).